ravenframework/SupervisedLearning/ROMCollection.py

```diff
@@ -1866,6 +1867,273 @@ def evaluate(self, edict):
+    # keys which include targets and other metadata
+    resultsKeys = [x for x in self.target]
+    resultsKeys.extend([self.pivotID,'_indexMap'])
+    results = {}  # this is the dictionary that gets sent upstream to the rest of RAVEN/ROM stuff
+
+    numMacro = len(self._macroSteps)
+
+    # step through trained macroSteps (e.g., years)
+    macroIndexValues = []
+    for m, (macroID, model) in enumerate(sorted(self._macroSteps.items(), key=lambda x: x[0])):
+      macroIndexValues.append(macroID)
+      decompModels = self._decompSteps[macroID]
+      decompResults = {x:{} for x in self.target if x != self.pivotID}
+      for d, (lvl, decompModel) in enumerate(decompModels.items()):
+        # evaluate algos for given decomposition level
+        subResult = decompModel.evaluate(edict) # TODO same input for all macro steps? True for ARMA at least...
+        # evaluate global algos for given decomposition level
+        decompResult = decompModel.finalizeGlobalRomSegmentEvaluation(None, subResult, weights=None, slicer=None)
+
+        # singular decomposition level stored in larger decompResult*s* dict
+        for target, contents in decompResult.items():
+          if target not in [self.pivotID,'_indexMap']:
+            decompResults[target][lvl] = contents
+
+      # combine results and save to macroStep model trainedParams
+      model._updateMRTrainedParams(decompResults)
+
+      # run first set of MR TSA algorithms (should include some sort of MRA transformer)
+      macroResults = model.evaluate(edict)
+
+      # if first macro step, must set up dict correctly
+      if not results:
+        for k in resultsKeys:
+          if k == self.pivotID:
+            # storing pivot parameter array (assuming same for all macroSteps/years)
+            results[k] = macroResults[self.pivotID]
+          elif k == '_indexMap':
+            # an instance of _indexMap stored for each target
+            results[k] = {}
+            for target in self.target:
+              if target != self.pivotID:
+                results[k][target] = [self._macroParameter, self.pivotID]
+          elif k in noPivotTargets:
+            # this was taken from the Interpolated method, specifying dtype fixed an error as well...
+            results[k] = np.zeros([numMacro] + list((macroResults[noPivotTargets[0]].shape)), dtype=object)
+
+      # storing results from a single macroStep (includes all decomp levels)
+      for target, contents in results.items():
+        if target not in [self.pivotID, '_indexMap']:
+          contents = macroResults[target]
+
+    results[self._macroParameter] = macroIndexValues
+    return results
+
+  def writeXML(self, writeTo, targets=None, skip=None):
+    """
+      Write out ARMA information
+      @ In, writeTo, xmlUtils.StaticXmlElement, entity to write to
+      @ In, targets, list, optional, unused
+      @ In, skip, list, optional, unused
+      @ Out, None
+    """
+
```

Comment:
remove?

Reply:
I found a way to loop and write stuff to the romMeta, this method is now filled in!